### DISCUSSION TITLE
org.eclipse.tycho.surefire.osgibooter java compatibility

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-tycho/tycho#1542] [General]
Currently `org.eclipse.tycho.surefire.osgibooter` claims to be java 1.5 compatible, but using a true 1.5 JVM reveals that some new code methods are used (mostly regarding strings and collection usage).

So the question is what should be the lowest minimal java version to support for testing?

**Comment by akurtakov** [comment]
Does it work with Java 8 ? If it works that should be fine min version. No one complaint so far so it should be fine to do this change. Furthermore finding Java 5 JVM is getting increasingly difficult.

**Reply by laeubi** [reply]
Actually it uses in some code path even Java 11 methods but that might have not been discovered so far if you not use certain options.

**Reply by laeubi** [reply]
As its seams we are about a new major release as discussed here:

https://github.com/eclipse-tycho/tycho/discussions/1528#discussioncomment-3888457

I will upgrade to 1.8 and add this simply to the release notes.

**Reply by laeubi** [reply]
See https://github.com/eclipse-tycho/tycho/pull/1543
